### ISSUE TITLE
WFS Pre-Defined test failing since release 2022.0

### ISSUE THREAD

**Issue by petbod-lm** [INSPIRE-MIF/helpdesk-validator#718]
**WFS pre-defined test failing since release 2022.0**

**WFS service URL:**  https://geo-inspire.trafikverket.se/mapservice/wfs.axd/TN_RoadTransportNetwork?ACCEPTVERSIONS=2.0.0&request=GetCapabilities&service=WFS&VERSION=2.0.0

**WFS Service metadata:** https://www.geodata.se/geodataportalen/srv/swe/csw-inspire?request=GetRecordById&service=CSW&version=2.0.2&elementSetName=full&outputSchema=csw:IsoRecord&id=84ce3866-7e61-4a30-9827-c51631ad0121

Since the release of ETF Validator v2022.0 on 19/01/2022 we see all the WFS service tests are failing due to the fix for issue #638 ([pull request 688](https://github.com/inspire-eu-validation/ets-repository/pull/688))  _Check operatesOn/xlink:href instead of operatesOn/uuidref_

I have updated several WFS service metadata to include the Resource ID of the dataset metadata in the operatesOn/xlink:href URL. 
However the validator continues to return an error in the step 'MD_Identifier exists in the ServiceMetadata'

The error message returns no useful information:
"The MD_identifier '{id}' is found in the metadata document but does not exist in the service Metadata. The service metadata refers to these identifiers: '{datasetIdentifiersList}'. Check the Service Metadata and the Metadata document."

Having updated the Service metadata this test should pass but I continue to see the test fail.

 
I have downloaded etf-repository-2022.0.zip and reviewed the test script service/ds-wfs-pre-defined-soapui-project.xml
A summary of the steps relating the error is below:

script ds-wfs-pre-defined-soapui-project.xml

`line 7074  <con:testStep name="Property Transfer Metadata Identifiers" ...`
extracts all operatesOn URLs from the service metadata and saves them to a comma-seperated list

`line 7100 <con:testStep type="groovy" name="Loop Metadata URLs from Service Metadata"`
finds all operatesOn URLs in the service metadata and saves them in a list
`line 7114 def metadataUrlNodes = serviceMDXmlHolder.getNodeValues("//*:operatesOn/@*:href");`

`line 7130 <con:testStep type="httprequest" name="Get Metadata from Service Metadata" id="da4394fa-c719-4018-8667-97114bf17cb1">`
calls the operatesOn URL and tests the reponse 

`line 7170 <con:assertion type="XQuery Match" name="MD_Identifier exist in the ServiceMetadata" id="f9d73158-df14-4818-86b3-fe04649351ca">`
extracts the resource ID from the metadata file //gmd:MD_DataIdentification/gmd:citation/gmd:CI_Citation/gmd:identifier/gmd:MD_Identifier/gmd:code/gco:CharacterString
and tests that the result is in the list created in step in line 7074

_(This test should compare the Resource ID in the dataset metadata with the URL used to retreive the metadata and not use the list of operatesOn URLs created on line 7074.
This would remove a step in the test and more accurately define the intention of this test)_
 
I have also checked the logs in our local ETF validator service and I see the following error: 
```
Error on line 8 column 14
  XPST0003: XQuery syntax error in #...-c7b6-4331-9c42-4bf7378cd9fb',#:
    No closing ';' found for entity or character reference
java.lang.Exception: Test step "Get Metadata from Service Metadata" failed!
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
        at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:77)
        at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrapNoCoerce.callConstructor(ConstructorSite.java:102)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:190)
        at de.interactive_instruments.xtf.ProjectHelper.runTestStep(ProjectHelper.groovy:419)
        at de.interactive_instruments.xtf.ProjectHelper.runTestStep(ProjectHelper.groovy)
        at de.interactive_instruments.xtf.ProjectHelper$runTestStep.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:108)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at Script1.run(Script1.groovy:20)
        at com.eviware.soapui.support.scripting.groovy.SoapUIGroovyScriptEngine.run(SoapUIGroovyScriptEngine.java:92)
        at com.eviware.soapui.impl.wsdl.teststeps.WsdlGroovyScriptTestStep.run(WsdlGroovyScriptTestStep.java:141)
        at com.eviware.soapui.impl.wsdl.support.AbstractTestCaseRunner.runTestStep(AbstractTestCaseRunner.java:211)
        at com.eviware.soapui.impl.wsdl.testcase.WsdlTestCaseRunner.runCurrentTestStep(WsdlTestCaseRunner.java:47)
        at com.eviware.soapui.impl.wsdl.support.AbstractTestCaseRunner.internalRun(AbstractTestCaseRunner.java:138)
        at com.eviware.soapui.impl.wsdl.support.AbstractTestCaseRunner.internalRun(AbstractTestCaseRunner.java:46)
        at com.eviware.soapui.impl.wsdl.support.AbstractTestRunner.run(AbstractTestRunner.java:129)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)

```
The value that is mentioned in the exception is the end of the last operatesOn URL in the service metadata.

**Comment by petbod-lm** [comment]
Is this fix ready in Staging?  
I have tested the WFS service every week in Staging since the 'ready for testing' status udapte on this ticket and continue to see the same error I reported.

**Comment by fabiovinci** [comment]
Dear @petbod-lm,

we are reviewing the ATS and the ETS because it seems that some tests, required by "TG Requirement 53" are not described in the ATS. Looking at the screenshot below, it seems only the first part of the requirement is implemented.

![image](https://user-images.githubusercontent.com/22212406/171434719-5e641317-5016-43f3-ad9a-69bfff9b5aef.png)

On the contrary, in the ETS there are some additional tests not required by "TG Requirement 53" and not described in the ATS.

![image](https://user-images.githubusercontent.com/22212406/171434173-e1847306-1192-4996-b6bb-6a5d1e0a5073.png)

The additional tests are related to a check on the linking between dataset and service and they take into consideration some recommendations of the TG MD, for this reason, they can be removed from the "Service Metadata" test case.

We will keep you posted.

**Comment by fabiovinci** [comment]
Dear @petbod-lm,

we fixed the error according to what was indicated in the mentioned issue (# 780).

The solution is available in the staging instance and it will be included in the next release 2022.2.
